### ISSUE TITLE
Create a User Interface with CAP (SAP HANA Cloud)

### ISSUE THREAD

**Issue by js1972** [SAPDocuments/Issues#264]
Tutorials: https://developers.sap.com/tutorials/hana-cloud-cap-create-ui.html
--------------------------

Once completed this step (as well as the following authentication step) I find that I cannot build the MTA without it going into an infinite loop and using up all the disk space in my BAS workspace.

To be able to build the MTA I had to add this to the ui deployer module:
```
build-parameters:
      ignore: [".*/"]
```

Now that I have a built MTA if I try and deploy it, it fails on the ui deployer.

```
Error uploading content module "MyHANAApp_ui_deployer" in service "MyHANAApp_html_repo_host" /// + heaps more
```

When I look at the ui deployer module in the mat file I see this:

```
  - name: MyHANAApp_ui_deployer
    type: com.sap.application.content
    path: .
    requires:
      - name: MyHANAApp_html_repo_host
        parameters:
          content-target: true
    build-parameters:
      ignore: [".*/"]
```

This does not seem right to me. It missing the bit about what .zip file to pick up that contains the built app? I have re-done the tutorial a second time and got exactly the same result. Any ideas what I could be doing wrong?
I also cant see any ui5 tooling yaml file which build the Fiori app?

Thanks, Jason.

**Comment by jung-thomas** [comment]
The first item you hit is a bug in the MBT tool that is known and being addressed. 

Building the MTAR and deploying the full project is not actually within the scope of this tutorial. We've tried to keep the deployment descriptor (mta.yaml) up to date in this tutorial but the focus on the development experience. There are other tutorials that cover the deployment process and that step might require additional adjustments to the mta.yaml beyond what the wizard has generated into the mta.yaml at this point.

**Comment by js1972** [comment]
oh. ok thanks or the update. Do you have a specific tutorial in mind on how to convert this to a deployable project? The other tutorials are structured quite differently without moving the html5 modules under the app folder (which doubles as the approuter).
I can easily add a ui5.yaml to build the Fiori app and adjust the package.json (of the Fiori app) to do that but that's where I get lost. Do we need to have a new module for the Fiori app in the MTA to make it deployable?

**Comment by jung-thomas** [comment]
I did add a new tutorial for the deployment steps.  I didn't adjust the project structure but instead the mta.yaml and used the build-parameters/ignore to optimize the output even with the CAP service running from the root. 

I did choose to remove the HTML5 repo UI deployer section and run the UI5 content directly from the Application Router. I wanted to simplify things for those that aren't used to the more advanced UI deployment options. But I tried to leave in some explanation of those other options.  

https://developers.sap.com/tutorials/hana-cloud-cap-deploy-mta.html
